test/src/unit-capi-group.cc

```diff
@@ -230,20 +230,46 @@ void GroupFx::vacuum(const char* group_uri) const {
 }
 
 std::vector<std::pair<tiledb::sm::URI, tiledb_object_t>> GroupFx::read_group(
-    tiledb_group_t* group) const {
+    tiledb_group_t* group, bool use_get_member_by_index_v2) const {
   std::vector<std::pair<tiledb::sm::URI, tiledb_object_t>> ret;
   uint64_t count = 0;
-  char* uri;
-  tiledb_object_t type;
-  char* name;
   int rc = tiledb_group_get_member_count(ctx_, group, &count);
   REQUIRE(rc == TILEDB_OK);
-  for (uint64_t i = 0; i < count; i++) {
-    rc = tiledb_group_get_member_by_index(ctx_, group, i, &uri, &type, &name);
-    REQUIRE(rc == TILEDB_OK);
-    ret.emplace_back(uri, type);
-    std::free(uri);
-    std::free(name);
+  if (use_get_member_by_index_v2) {
+    for (uint64_t i = 0; i < count; i++) {
+      std::string uri;
+      tiledb_object_t type;
+      tiledb_string_t *uri_handle, *name_handle;
+      rc = tiledb_group_get_member_by_index_v2(
+          ctx_, group, i, &uri_handle, &type, &name_handle);
+      REQUIRE(rc == TILEDB_OK);
+      const char* uri_str;
+      size_t uri_size;
+      rc = tiledb_string_view(uri_handle, &uri_str, &uri_size);
+      CHECK(rc == TILEDB_OK);
+      uri = std::string(uri_str, uri_size);
+      rc = tiledb_string_free(&uri_handle);
+      CHECK(rc == TILEDB_OK);
+      if (name_handle) {
+        rc = tiledb_string_free(&name_handle);
+        CHECK(rc == TILEDB_OK);
+      }
+      ret.emplace_back(uri, type);
+    }
+  } else {
+    // When tiledb_group_get_member_by_index gets removed, the else part can
+    // simply be removed.
+    for (uint64_t i = 0; i < count; i++) {
+      std::string uri;
+      tiledb_object_t type;
+      char *uri_ptr, *name_ptr;
+      // NOTE: The following function leaks memory.
```

Comment:
Can you expand on this comment? Is the leak that the `uri_ptr` and `name_ptr` are the result of allocations and should be freed? I'd very much not elide those steps because folks will look at the test code for proper use and if they're not paying attention while the deprecated versions still exist we'll have led them astray.

Reply:
In the past when I was debugging a program that was using the prebuilt TileDB binaries, I had ran into the situation where I was calling `Group::member` and the debug CRT freed memory the release CRT was allocating, causing a crash.

But on second thought it's very unlikely for the unit tests to mix the CRTs. I added the calls to `free` back.